cache/cache.go

```diff
@@ -0,0 +1,49 @@
+package cache
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sync"
+
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+type CacheKey interface {
+	string
+}
+
+type CacheValue interface {
+	string | options.IAMRoleOptions
+}
+
+type GenericCache[cacheValue CacheValue] struct {
+	Cache map[string]cacheValue
+	Mutex *sync.Mutex
+}
+
+// NewStringCache - create new generic cache
```

Comment:
I think should be `NewGenericCache`, to match function name...

Reply:
Thanks for the correction